provider/doc.go

```diff
@@ -0,0 +1,3 @@
+// Package provider contains the interface and primary types required to
+// implement a Terraform resource provider.
+package provider
```

Comment:
While I quite like the look of `provider.Interface`, this package name also squats on the variable name we usually use for provider instances.

`providers.Interface` doesn't quite have the same nice ring to it, but it still seems clear enough and allows us to keep calling our variables `provider`. What do you think?

Reply:
Yeah, I thought about that, but in reality we rarely use the identifier `provider`. 
There are still a few places though, so I'll change it over since any saved work right now is probably worth it.